extensions/wikia/AdEngine/AdEngine2ContextService.class.php

```diff
@@ -17,8 +17,11 @@ public function getContext( Title $title, $skinName ) {
 			$adPageTypeService = new AdEngine2PageTypeService();
 			$wikiaPageType = new WikiaPageType();
 			$pageType = $wikiaPageType->getPageType();
-			$hasFeaturedVideo = !empty( $wg->EnableArticleFeaturedVideo )
-				&& ArticleVideoContext::isFeaturedVideoEmbedded( $title->getArticleID() );
+			$articleId = $title->getArticleID();
+			$hasFeaturedVideo = ( !empty( $wg->EnableArticleFeaturedVideo )
+				&& ArticleVideoContext::isFeaturedVideoEmbedded( $articleId ) ) ||
+				// XW-4713 | Hide UAPs on pages running the Recommended Video ABTest
+				ArticleVideoContext::isRecommendedVideoAvailable( $articleId );
```

Comment:
@bart IMO it should be`!empty( $wg->EnableArticleFeaturedVideo ) && (ArticleVideoContext::isFeaturedVideoEmbedded( $articleId ) || ArticleVideoContext::isRecommendedVideoAvailable( $articleId )); ` since ArticleVideoContext class is not loaded if wgFeaturedVideo is not enabled